pyleecan/Methods/Output/LUTdq/interp_Ploss_dqh.py

```diff
@@ -33,23 +35,15 @@ def interp_Ploss_dqh(self, Id, Iq, N0):
 
     felec = N0 / 60 * p
 
-    type_skin_effect = self.simu.loss.type_skin_effect
-    Tsta = self.simu.loss.Tsta
-
-    Ploss_dqh = np.zeros((len(self.output_list), 5))
+    Ploss_dqh = np.zeros(
+        (len(self.output_list), len(self.output_list[0].loss.loss_list))
+    )
     for ii, out in enumerate(self.output_list):
         OP = out.elec.OP.copy()
         OP.felec = felec
-        Ploss_dqh[ii, 0] = comp_loss_joule(
-            lam=self.simu.machine.stator,
-            OP=OP,
-            T_op=Tsta,
-            type_skin_effect=type_skin_effect,
-        )
-        Ploss_dqh[ii, 1] = out.loss.get_loss_group("stator core", felec)
-        Ploss_dqh[ii, 2] = out.loss.get_loss_group("rotor magnets", felec)
-        Ploss_dqh[ii, 3] = out.loss.get_loss_group("rotor core", felec)
-        Ploss_dqh[ii, 4] = out.loss.get_loss_group("stator winding", felec)
+        for kk, loss in enumerate(out.loss.loss_list):
+            if loss.loss_model not in exclude_models:
+                Ploss_dqh[ii, kk] = loss.get_loss_scalar(felec)
```

Comment:
I think Ploss_dqh definition (see Parameter Returns) isn't valid any more, i.e. order and assosiated losses depend on out.loss.loss_list now.
Further Ploss_dqh_mat need to have a dynamic size for dim == 3 (see Line 54).